book/src/todo_app_1.md

```diff
@@ -0,0 +1,231 @@
+As usual, we have to list all [ancestors](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#ancestors) and [interfaces](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#implements) apart from `GObject` and `GInitiallyUnowned`.
+
+<span class="filename">Filename: listings/todo_app/1/window/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/mod.rs:glib_wrapper}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Then we initialize the composite template for `imp::Window`.
+We store references to the entry, the list view as well as the model.
+This will come in handy when we later add methods to our window.
+After that, we add the typical boilerplate for initializing composite templates.
+We only have to assure that the `class` attribute of the template in `window.ui` matches `NAME`.
+
+<span class="filename">Filename: listings/todo_app/1/window/imp.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/window/imp.rs:struct_and_subclass}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+`main.rs` also does not hold any surprises for us.
+
+<span class="filename">Filename: listings/todo_app/1/main.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/main.rs:main}}
+```
+
+So far so good.
+The main user interface is done, but the entry does not react to input yet.
+Also, where would the input go?
+We have not even set up the list view yet.
+Let us take care of that!
+
+As discussed in the [lists chapter](./lists.html),
+we start out by creating a custom GObject.
+This object will store the state of the task consisting of:
+- a boolean describing whether the task is completed or not, and
+- a string holding the content.
+
+<span class="filename">Filename: listings/todo_app/1/todo_object/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/todo_object/mod.rs:glib_wrapper_and_new}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Unlike the lists chapter, the state is stored in a struct rather than in individual members of `imp::TodoObject`.
+
+<span class="filename">Filename: listings/todo_app/1/todo_object/mod.rs</span>
+
+```rust
+{{#rustdoc_include ../listings/todo_app/1/todo_object/mod.rs:todo_data}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Not only that, we see that it is also wrapped in a [`Rc`](https://doc.rust-lang.org/std/rc/struct.Rc.html).
+This is not really relevant for now, but will be very convenient when we want to save the state in one of the following chapters.
+Exposing `completed` and `content` as properties does not become much different that way, so we will not discuss it further.
+If you are curious, you can press on the small eye symbol on the top right of the code snippet to read the implementation.
```

Comment:
This sentence can also go if previous one is deleted I think

Reply:
I at least have to mention, that the member variables are exposed as properties.
Because that's how I bind `TodoRow` to `TodoObject` :)